kani_metadata/src/cbmc_solver.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use serde::{Deserialize, Serialize};
+use strum_macros::{AsRefStr, EnumString, EnumVariantNames};
+
+/// An enum for CBMC solver options. All variants are handled by Kani, except for
+/// the "Custom" one, which it passes as is to CBMC's `--external-sat-solver`
+/// option.
+#[derive(
+    Debug,
+    Clone,
+    AsRefStr,
+    EnumString,
+    EnumVariantNames,
+    PartialEq,
+    Eq,
+    Serialize,
+    Deserialize
+)]
+#[strum(serialize_all = "snake_case")]
+pub enum CbmcSolver {
```

Comment:
What's the reason for this to be `CbmcSolver` and not just `Solver`?

Reply:
Since the solver is currently specific for CBMC, and we're not yet clear on future plans, and whether this solver will apply to other engines as well, I added CBMC to its name.

Reply:
I tend to agree with @adpaco-aws here, but this is an implementation detail that is internal to our code, so we can easily rename this later.